torchmetrics/image/ssim.py

```diff
@@ -164,9 +180,9 @@ class MultiScaleStructuralSimilarityIndexMeasure(Metric):
 
     def __init__(
         self,
-        kernel_size: Sequence[int] = (11, 11),
-        sigma: Sequence[float] = (1.5, 1.5),
-        reduction: str = "elementwise_mean",
+        gaussian_kernel: bool = True,
+        kernel_size: Union[int, Sequence[int]] = 11,
+        sigma: Union[float, Sequence[float]] = 1.5,
```

Comment:
can we reorder to `kernel_size`, `sigma`, `gaussian_kernel`?
what happened to the reduction parameter? that need to be deprecated if it is being removed

Reply:
That ordering works for me.

Concerning the reduction parameter: Before, there was no option of no reduction. However, it definitely is necessary to have access to the metric for every image in the batch. 
I wasn't sure what is "standard" in this package for the reduction parameter. If you wish to have this parameter, we can add it again. I also opened this up to discussion somewhere else in this PR.

Reply:
Yep, exactly, with `reduction=None` that function returned the full SSIM images. However, the SSIM metric is a scalar metric, i.e., `reduction=None` should return one scalar per image. This was not available.

Reply:
You can have a look at the skimage implementation: https://github.com/scikit-image/scikit-image/blob/main/skimage/metrics/_structural_similarity.py#L15-L254
A scalar (cropped and mean SSIM image) is returned. In my opinion, this should also be the default here (of course in the batched version).

Reply:
Note that the SSIM metric should not exactly the same as the mean of the SSIM image. The SSIM image has the same size as the input image. However, the SSIM metric needs to be calculated on a cropped version in order to avoid border effects

Reply:
I agree that we should more or less match the skimage implementation. I just want to make sure that if I call:
```python
structural_similarity_index_measure(preds, target)
```
on a batch of images, then it will return a scalar and not vector equal to the batch size (also to keep things backwards compatible). The final reduction is what I want the `reduction` parameter to take care of.

Reply:
Adding to @SkafteNicki : the `reduction` is the reduction across the batch dimension. Meaning that if that one is `none` it would be equal to having one value per sample in the batch. I think it should default to some averaging to get a single scalar for a whole batch